modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/SortAggregateNode.java

```diff
@@ -159,6 +161,10 @@ public void end() throws Exception {
 
         waiting = -1;
 
+        if (grp == null && accFactory != null) {
```

Comment:
this will produce empty row when none is expected in query like `SELECT min(c) FROM t GROUP BY a` when table is empty. BTW, we need execution test for this case

to initialise group properly, please, take a look at `org.apache.ignite.internal.sql.engine.exec.rel.HashAggregateNode.Grouping#init`

Reply:
got it!

Reply:
let's add _execution_ test for this case